src/SequenceKit.ts

```diff
@@ -0,0 +1,53 @@
+// Author: Pouya Kary <k@karyfoundation.org> (in case you had any questions...)
+
+import {colors} from "./views/css/colors";
+import {Suggestion} from "./plugins/autocompletion_utils/Common";
+
+export interface SortableObject {
+    weight: number;
+    value: Suggestion;
+}
+
+/**
+ * Checks if the __key__ exists as a _sequence_ in the __element__.
+ */
+export function SequenceFilter(element: string, key: string): boolean {
```

Comment:
Can we make this function return an array of indices of matched letters? An empty array would mean `false`, and we could reuse those indices to highlight letters instead of implementing the algorithm twice.